### DISCUSSION TITLE
PCB question

### DISCUSSION THREAD

**Discussion by bkbartk** [TillFleisch/ESPHome-Philips-Smart-Coffee#58] [Q&A] [answered]
Hello,

My project is stuck on the part where I still need to fix some soldering(for a few months now),
now I just saw your new PCB design
https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee?tab=readme-ov-file#custom-pcb
which I think would be ideal for me as that saves me the soldering part.

but I have 2 questions,
1 where can I order this pcb (or does this depend on where I live)
2 is it possible to serial flash without soldering if I accidentally brick the esp?

**Comment by TillFleisch** [comment, marked as answer]
> 1 where can I order this pcb (or does this depend on where I live)

I ordered my PCBs from [JLCPCB](https://jlcpcb.com/) and the parts from [LCSC](https://www.lcsc.com/) and then assembled them by hand in my office. Both packages shipped from their warehouses to Germany without problems. JLCPCB also offers PCB manufacturing and assembly in one. I cannot say anything regarding this, their quality, or pricing. [PCBWay](https://www.pcbway.com/) offers similar services as far as I know. There are probably plenty of resources comparing the two. Please also consider local service providers if possible.

> 2 is it possible to serial flash without soldering if I accidentally brick the esp?

Since the board design does not include a serial to USB converter you need an external USB to serial adapter. On the back side of the PCB there Testpads which can be used to flash firmware onto the ESP8266-12F module. GPIO0/P0 must be connected to ground (during power-up), as with any other ESP8266-based design, to put the ESP8266 into programming mode. RX/TX can then be used for communication. The Testpads can be probed with pogo-pins in a small test-jig or male dupont connectors held manually.

**Comment by bkbartk** [comment]
Thank for the reply,
At first I was thinking you might be shipping those yourself making a little profit.

Ordering the pcb and parts separate wouldn't help me saving the part where I need to Solder,
I know next to nothing about pcb development so my questions might be dumb.
When I try to get an estimata on JLCPCB it's only per 5 and for assembly it needs some kind of CPL file.
I assume this file specifies where the parts should be located for assembly. but I'm not sure,
Is there a file like this available?
When I try to use PCBWay, the questions are somehow less clear, and even before uploading the BOM I already had an order instead of a quote. so I cancelled that process quickly. I noticed however I needed to upload a "centroid file", I think that one might is the equilivant of a CPL file.

I noticed the testpad on the back, but I am a bit afraid it's hard to keep male dupont connentors on place long enough to flash, but it's worth a try. for most bourds I flashed there were this little holes just like in J1 and J2 where I it easy to hold the male dupont pins, but I will give it a try, and else I have to order a test-jig.
I was kind of hoping flashing would be possible through J1 or J2 but If not I will find it out.

**Comment by TillFleisch** [comment]
> At first I was thinking you might be shipping those yourself making a little profit.

Currently not possible since that would be taxable income (even if I didn't make any profit), which requires special registration and paperwork above a certain threshold.

> it needs some kind of CPL file.

I have opened the shared project on oswhlab.com and exported a "Pick and Place" csv-file. I believe this is what's required as the CPL file. I have also added it as an attachment to the oswhlab page. Since I have not been through the PCB assembly process I can't say much more about it. (The Gerber and BOM files were generated similary using their online editor)

>  I was kind of hoping flashing would be possible through J1 or J2 but If not I will find it out.

RX/TX are not exposed via the top headers as they are used in the picoflex connectors, therefore they would be kind of useless as pins. For debugging the Testpads are sufficient. Alternatively, if the testpads don't work for you, you could get a spare picoflex cable and cut that in half to access the pins (GND,5V, RX,TX) via the mainboard picoflex connector.
Or you could solder a thin wire to either the Testpads or the pins from the mainboard picoflex connector (bottom side). Be careful when soldering to the testpads as they can come loose from the board.

**Comment by bkbartk** [comment]
> I have opened the shared project on oswhlab.com and exported a "Pick and Place" csv-file

This file seems to be accepted at both [JLCPCB](https://jlcpcb.com/) and [PCBWay](https://www.pcbway.com/)

at JLCPCB 2 part are not in stock 1 of which are the picoflex connector, so that's a shame. so I tried PCBWay and have some kind of provisional quote, but I have the feeling this is without parts, but I will see how it goes. I really like this pcb design, 

>  you could get a spare picoflex cable and cut that in half to access the pins (GND,5V, RX,TX) via the mainboard picoflex connector

I already have a few a this so that is a really good solution, never thought about the fact that I can use 5V for flashing esp's

**Comment by TillFleisch** [comment]
You cannot use 5V for flashing an ESP. If you connect 5V to the ESP module it will most likely get fried. The circuit board design includes a voltage regulator which lowers the 5V input to 3.3V required by the ESP.

You may find a different seller/provider for the picoflex connectors (using the part number 90325-8 or 902350008). Given that the part number (the one from molex) is the same, the parts should be identical. Alternatively you could source them from somewhere else and install them yourself (this requires soldering :/ ).

**Reply by deWasbeer** [reply]
So I can flash the board by connecting the wires to the GND,5V, RX,TX of the testpads?

**Comment by bkbartk** [comment]
thanks,
I ordered at JLCPCB and replaced "RC0805JR-0710KL" by "RMS10FT1002" as they didn't have enough in stock.

PCBWay has a few issues but I needed to contact support to get the details, so I didn't continue on that road. 

There was no possible way to replace the picoflex connector(I think Molex has some kind of patent making it hard to aquire) but I still have a few of those connectors here on earlier attempts, So it means some soldering, but if all goes well not to much.
for the rest I left everything to the default.

the pcb's will be delivered in 12 to 14 days as that saved me a few $. 

I keep you informed in case others might have questions regarding the process.

**Reply by ntschetterNY** [reply]
Thanks for the tip above - I was able to sub out the part number as you did with JLCPCB, however they came back with the below question on the "Main Board" & "Display." Did you receive this feedback from them, and were you able to resolve? Attached is their message, with screenshot,
[Gmail - PCBA order issue of SMT02406271810029_Y1 8017046A ----JLCPCB.pdf](https://github.com/user-attachments/files/16045877/Gmail.-.PCBA.order.issue.of.SMT02406271810029_Y1.8017046A.----JLCPCB.pdf)
 in text form below. 

- As shown below, seems the packages of the parts (MAINBOARD、DISPLAY ) cannot match the corresponding pads on the PCB board, so we are unable to assemble them. For this case there are 3 options for your choose as follows:
- A- we enable the replace part button for you that you can re-select the corrected part
- B- we leave this parts unpopulated and refund cost of the mismatch parts back to you to proceed and can’t ship the unsoldered parts with PCB due to the customs clearance issue,
- C-cancel this PCB order and SMT order to place new order (there is no replace gerber file button for SMT order)

If anyone is located close to NY, USA and has a spare board i'd be glad to make it worthwhile to send one over.

**Reply by bkbartk** [reply]
It looks like you are trying to replace the picoflex connector with a resistor,
that won't work
I replaced 1 resistor with another one. the picoflex connector I had still laying around from my attempts to use a wemos.
If they don't have to picoflex you should try to find those somewhere else and solder them on.
it's hard to get those. 

edit:
https://nl.farnell.com/en-NL/molex/90814-0908/connector-header-8pos-1row-1-27mm/dp/2779812

**Reply by aneisch** [reply]
@ntschetterNY I just ordered some of https://oshwlab.com/tradstaat/coffeepcb and should have them on hand and assembled in the next month or so (including populated d1 mini with stub config). I planned to offer the extras at cost to others in the US. I can reach out when they're ready if you're interested.

**Reply by ntschetterNY** [reply]
@bkbartk - thanks for the feedback! Confirmed that's what I was trying to do. Appreciate the help. I have now 5 PCB's without the picoflex connector, do you have use for any / in USA?
@aneisch - PCB's came in from JLPCB without the picoflex, i'd definitely be interested in picking one up when you have them assembled. Thx! Same offer above stands if you have use for a PCB without the picoflex.

**Reply by aneisch** [reply]
@ntschetterNY can you email me? Address is in my profile.

**Comment by bkbartk** [comment]
Just an update,
Today I received my ordered PCB's for JLCBCP
![image](https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/assets/17070805/6f68a6f6-2e30-48aa-8793-1c86a1f719b4)
I missed the fact that a header was pre-solered, but that made flashing a little easier.
![image](https://github.com/TillFleisch/ESPHome-Philips-Smart-Coffee/assets/17070805/ead6d1ff-4495-400d-baf5-7092d45834f1)
after this, I needed to solder 3 times( don't ask, I'm just not good at it) and now it works.
hope it keeps working 

Thank you very much for this project.

**Reply by TillFleisch** [reply]
Happy to hear everything worked out! The green solder mask looks pretty neat.

**Reply by CamelCase1072** [reply]
@bkbartk @TillFleisch: I was also planning on ordering a PCB but seeing that you got both ordered mulitple pieces would you be willing to sell one and mail it to Germany for a fair price?

**Reply by bkbartk** [reply]
for me that would mean international shipping as I don't live in Germany,
this would bring the total price for parts and shipping around ~€20, - €25,- for just one peace.
if you order at https://jlcpcb.com/ you can have 5 parts including shipment to Europe(at least The Netherlands) for around ~$68,-

also the boards I ordered are without the Picoflex connectors, as they were out of stock, you have to solder them on yourself.

**Reply by Marquo1** [reply]
@MEfinger I also wanted to make the PCB designed by Till, but it was too expensive to have it pre-assembled and I don't have time to learn SMD soldering to bring down the cost. 

But the other option he mentioned with non SMD components is much cheaper. 

I just ordered 5 PCB's from JLPCB for 4 eur incl shipping. Tweaked design to remove extra connectors as I don't need them.
The rest is a Wemos D1 (5 eur) and a transistor (2 eur for 10 pcs.) - both from Ali. The picoflex connector and cable are 5 eur on Conrad (+ 7 shipping cost, unfortunately). 
In the end it's 22 eur if you're ordering on your own, which is much more acceptable. Some soldering required (but it's always required for picoflex).

**Reply by knschuckmann** [reply]
@Marquo1 can you provide the link to your pcb setup

**Comment by ntschetterNY** [comment]
That would be great! I'll likely take you up on that, I can send the PCB's
that I have coming without the picoflex connector too (have 5 coming next
week from JLPCB).

**Reply by aneisch** [reply]
@ntschetterNY can you email me? Address is in my profile.
![PXL_20240720_195453644 MP](https://github.com/user-attachments/assets/b86be353-240f-4bfc-8f0f-ecfedc7e9f62)

**Comment by f2103** [comment]
Does anybody have some assembled PCBs lying around in Europe (prefarably Germany) willing to sell?
Thx

**Reply by m4sb5r** [reply]
I would also be interested (Netherlands). I would also be willing to place an order and ship them around Europe. But I don't have experience with PCBs. I loaded all the files on JLCPCB and got the following two remarks:

The first remark:

>The comment (HB-PH3-25416PB2GOP) of this part does not match the one (HEADER_6) provided in your BOM. Please confirm

Can anybody explain what this means?

The second remark:

>There may be multiple types of parts, but one type of part has been matched. Please check.

This was about the MOLEX 903250008. I'm assuming that this is the picoflex connector others are referring to. Could I order this part from some place else than Farnell since the minimum order quantity is 600.

**Reply by Marquo1** [reply]
For Picoflex try Conrad -> [link](https://www.conrad.com/en/p/molex-pin-strip-standard-total-number-of-pins-8-contact-spacing-1-27-mm-903250008-1-pc-s-bag-2450721.html?srsltid=AfmBOookoptJUH5HtFzklN4Pn_Q0rykN9KNsWCuRN3OAg1T25tXXf3Du)
They also have matching cables, search for Molex part nr. 923150810.

**Comment by Llenas09** [comment]
Does anyone have PCBs assembled in Europe (preferably Spain) and want to sell them? Thanks

**Comment by kallejoken** [comment]
Does anyone have one left, even unequipped, within Germany?

**Reply by Marquo1** [reply]
Hi, 
I have a few unpopulated boards made for Wemos D1 (based on [this design](https://oshwlab.com/tradstaat/coffeepcb)).
I'm in the Netherlands.
I can send it to you for the cost of postage, if you let me know if it works (I don't have time to solder it so I never found out).

**Reply by kallejoken** [reply]
> Hi, I have a few unpopulated boards made for Wemos D1 (based on [this design](https://oshwlab.com/tradstaat/coffeepcb)). I'm in the Netherlands. I can send it to you for the cost of postage, if you let me know if it works (I don't have time to solder it so I never found out).

hi, yes I would like 1-2 pieces if that's okay with you.

**Reply by Llenas09** [reply]
> Hi, I have a few unpopulated boards made for Wemos D1 (based on [this design](https://oshwlab.com/tradstaat/coffeepcb)). I'm in the Netherlands. I can send it to you for the cost of postage, if you let me know if it works (I don't have time to solder it so I never found out).

Hi, can you send one board to Spain? Thanks!

**Reply by kallejoken** [reply]
> Hallo, ich habe ein paar unbestückte Platinen für Wemos D1 (basierend auf [diesem Design](https://oshwlab.com/tradstaat/coffeepcb) ). Ich bin in den Niederlanden. Ich kann sie Ihnen gegen Portokosten zusenden, wenn Sie mir Bescheid geben, ob sie funktionieren (ich habe keine Zeit zum Löten, also habe ich es nie herausgefunden).

How can you be reached?

**Reply by Marquo1** [reply]
You can send me an email - my adress is my github username on Google mail.

@kallejoken @Llenas09

**Comment by SchmDan** [comment]
Hi,

I've ordered the custom PCBs but had to order at least 5 pieces. I still got 4 PCBs left. If you need a PCB for your project just let me know or reach out to me over this advertisement (https://www.kleinanzeigen.de/s-anzeige/esphome-philips-smart-coffee-pcb/3063712265-168-5467). I'm based in Germany.

**Reply by FreeKillHD** [reply]
Hello, do you still have a PCB available?

**Comment by FreeKillHD** [comment]
Hello, I would also be interested in a fully soldered PCB. I live in Germany.

**Reply by knschuckmann** [reply]
I am also still waiting for mine to arrive but I can contact you when they are get here

**Reply by FreeKillHD** [reply]
Yes, that would be very nice

**Reply by TheGiantSwede** [reply]
@knschuckmann did you order 5?
if so how much do you want for one? and shipping to sweden?

**Reply by m4sb5r** [reply]
If you have any left over I would also be interested. I am located in the Netherlands.

**Reply by FreeKillHD** [reply]
I have bought today 5 pieces 
I can send in the EU

**Reply by TheGiantSwede** [reply]
How much? 😊

**Reply by FreeKillHD** [reply]
I would be Test the Board 
After that i information you

**Reply by TheGiantSwede** [reply]
> I would be Test the Board After that i information you

☕🍻

**Reply by m4sb5r** [reply]
> I have bought today 5 pieces I can send in the EU

I am also interested :).

**Reply by FreeKillHD** [reply]
I have now 4 left
I have Tested one of Them and its work
Picoflex Connectors are Not soldered![image](https://github.com/user-attachments/assets/bdfe9842-c1fe-4091-a0f6-63c3f9964d73)

**Reply by TheGiantSwede** [reply]
how much to buy one and shipping to sweden? :)

**Reply by FreeKillHD** [reply]
I’d like to sell them for €20 each. Why that price? Because I had to cover taxes and high shipping costs myself. Shipping to Sweden costs €6.99 without tracking or insurance, whereas options with tracking and insurance start at €14.49.

**Reply by TheGiantSwede** [reply]
sure! i bite for 27€
:)

**Reply by FreeKillHD** [reply]
> sure! i bite for 27€ :)

How would you like to pay?

**Reply by FreeKillHD** [reply]
> sure! i bite for 27€ :)

If you have Discord you can add me: xixslimshadyxix

**Reply by m4sb5r** [reply]
> I’d like to sell them for €20 each. Why that price? Because I had to cover taxes and high shipping costs myself. Shipping to Sweden costs €6.99 without tracking or insurance, whereas options with tracking and insurance start at €14.49.

If you have enough left over I would like to purchase 2 and have them shipped to the Netherlands. Let me know what that would cost and your preferred method of payment. I have both Revolut and Paypal. Thanks!

**Reply by FreeKillHD** [reply]
> > I’d like to sell them for €20 each. Why that price? Because I had to cover taxes and high shipping costs myself. Shipping to Sweden costs €6.99 without tracking or insurance, whereas options with tracking and insurance start at €14.49.
> 
> If you have enough left over I would like to purchase 2 and have them shipped to the Netherlands. Let me know what that would cost and your preferred method of payment. I have both Revolut and Paypal. Thanks!

I Sell it for You 47€ with Shipping for 2 Pieces

**Comment by c0ff3e** [comment]
i would also like to buy a built one.
live in Sweden.

Den fre 12 juni 2026 16:48FreeKillHD ***@***.***> skrev:

**Comment by c0ff3e** [comment]
I am would like to buy one, if i can pay via revolut or PayPal

Den tors 13 aug. 2026 19:29m4sb5r ***@***.***> skrev:

**Comment by c0ff3e** [comment]
get 2 and we sort it out in Sweden :)

Den ons 2 sep. 2026 17:02TheGiantSwede ***@***.***> skrev:

**Reply by FreeKillHD** [reply]
Are you Interested?

**Comment by c0ff3e** [comment]
I would also.  could be sweet if we might just need one shipment to Sweden.
drakepake @ dis

Den ons 2 sep. 2026 17:17FreeKillHD ***@***.***> skrev:

**Comment by FreeKillHD** [comment]
I have 3 Boards left

**Reply by yogibaerj-hub** [reply]
Hallo @FreeKillHD, hast du noch eine fertig bestückte Platine übrig? Ich wohne in Köln und würde gerne eine kaufen, für meine EP2231. Wenn ja, was möchtest du inklusive Versand haben? Bezahlen kann ich per PayPal. Danke dir!

**Reply by FreeKillHD** [reply]
> Hallo @FreeKillHD, hast du noch eine fertig bestückte Platine übrig? Ich wohne in Köln und würde gerne eine kaufen, für meine EP2231. Wenn ja, was möchtest du inklusive Versand haben? Bezahlen kann ich per PayPal. Danke dir!

Ja habe noch 3 insgesamt übrig 
Mit Versand 22€

**Reply by yogibaerj-hub** [reply]
> > Hallo @FreeKillHD, hast du noch eine fertig bestückte Platine übrig? Ich wohne in Köln und würde gerne eine kaufen, für meine EP2231. Wenn ja, was möchtest du inklusive Versand haben? Bezahlen kann ich per PayPal. Danke dir!
> 
> Ja habe noch 3 insgesamt übrig Mit Versand 22€

Super, 22 Euro passt.  Und zwei kurze Fragen noch: Sind die beiden Picoflex Stecker schon drauf gelötet, und ist schon eine Firmware auf dem ESP? Wenn ja, schick mir gerne deine PayPal Adresse, dann überweise ich.

**Reply by yogibaerj-hub** [reply]
Falls nein, würde ich das auch bezahlen wollen (wenn du dich bereit erklärst)

**Reply by FreeKillHD** [reply]
> > > Hallo @FreeKillHD, hast du noch eine fertig bestückte Platine übrig? Ich wohne in Köln und würde gerne eine kaufen, für meine EP2231. Wenn ja, was möchtest du inklusive Versand haben? Bezahlen kann ich per PayPal. Danke dir!
> 
> > 
> 
> > Ja habe noch 3 insgesamt übrig Mit Versand 22€
> 
> 
> 
> Super, 22 Euro passt.  Und zwei kurze Fragen noch: Sind die beiden Picoflex Stecker schon drauf gelötet, und ist schon eine Firmware auf dem ESP? Wenn ja, schick mir gerne deine PayPal Adresse, dann überweise ich.

Hast du Discord?
Dann könnte ich dir das etwas erklären 

Kannst mich gerne hinzufügen: xixslimshadyxix
